### ISSUE TITLE
Service discovery is not idempotent when nodes overlap

### ISSUE THREAD

**Issue by dstokesf5** [F5Networks/f5-appsvcs-extension#461]
### Environment
 * Application Services Version: 3.28.0
 * BIG-IP Version: 16.0.1.1

### Summary
A declaration using Consul service discovery is not idempotent when a node from Consul overlaps with a previously declared node.

### Steps To Reproduce
1. Submit the following declaration. It sets up two things. First is a Consul mock to point service discovery at, the second is an application with a statically defined pool member. The Consul mock will report the same node as the statically defined pool member:
```json
{
    "class": "ADC",
    "schemaVersion": "3.7.0",
    "setup": {
        "class": "Tenant",
        "consulMock": {
            "class": "Application",
            "vip": {
                "class": "Service_HTTP",
                "virtualAddresses": [
                    "192.0.2.0"
                ],
                "virtualPort": 8082,
                "iRules": [
                    {
                        "use": "consul_mock_irule"
                    }
                ]
            },
            "consul_mock_irule": {
                "class": "iRule",
                "iRule": {
                    "base64": "d2hlbiBIVFRQX1JFUVVFU1QgewogICAgc2V0IHJlc3BvbnNlIHtbewoJIklEIjogIjQ0RDJFOTk3LTI0NjItNDEwMy1COEYwLTQyOUUwRDgzMTcyRiIsCgkiTm9kZSI6ICIxOTIuMC4yLjEiLAoJIkFkZHJlc3MiOiAiMTkyLjAuMi4xIgp9XQp9CiAgICBIVFRQOjpyZXNwb25kIDIwMCBjb250ZW50ICIkcmVzcG9uc2UiIENvbnRlbnQtdHlwZSAiYXBwbGljYXRpb24vanNvbiIKfQ=="
                }
            }
        },
        "conflict": {
            "class": "Application",
            "pool": {
                "class": "Pool",
                "members": [
                    {
                        "shareNodes": true,
                        "serverAddresses": [
                            "192.0.2.1"
                        ],
                        "servicePort": 30036
                    }
                ]
            }
        }
    }
}
```

2. Submit a second declaration that makes use of the Consul mock:
```json
{
    "class": "ADC",
    "schemaVersion": "3.7.0",
    "controls": {
        "traceResponse": true
    },
    "tenant": {
        "class": "Tenant",
        "application": {
            "class": "Application",
            "vip": {
                "class": "Service_HTTP",
                "virtualAddresses": [
                    "192.0.2.3"
                ],
                "virtualPort": 443,
                "pool": "pool"
            },
            "pool": {
                "class": "Pool",
                "members": [
                    {
                        "servicePort": 30033,
                        "addressDiscovery": "consul",
                        "autoPopulate": true,
                        "uri": "http://192.0.2.0:8082/catalog/nodes",
                        "updateInterval": 5,
                        "rejectUnauthorized": false,
                        "shareNodes": true
                    }
                ]
            }
        }
    }
}
```

3. Submit the second declaration a second time.
4. Observer the following error response:
```json
    "results": [
        {
            "code": 422,
            "message": "declaration failed",
            "response": "01070110:3: Node address '/Common/192.0.2.1' is referenced by a member of pool '/tenant/application/pool'.",
            "host": "localhost",
            "tenant": "tenant",
            "runTime": 2128
        }
    ],
```

### Expected Behavior
The second posting of the second declaration should result in a "no change" result.

### Actual Behavior
Based on the trace, it looks like AS3 thinks it should be deleting `/Common/192.0.2.1` instead of doing nothing.

**Comment by 0xHiteshPatel** [comment]
One additional error scenario:

Removing the service discovery config from the `members` attribute of the second declaration results in a different failure.  Re-posting the same declaration again results in a success

1st POST Result - Error: 
```
    "results": [
        {
            "code": 422,
            "message": "declaration failed",
            "response": "01020036:3: The requested Pool Member (/tenant/application/pool /Common/192.0.2.1 30033) was not found.",
            "host": "localhost",
            "tenant": "tenant",
            "runTime": 6688
        }
    ],
```

2nd POST Result - Success:
```
    "results": [
        {
            "code": 200,
            "message": "no change",
            "host": "localhost",
            "tenant": "ISS232b",
            "runTime": 2685
        }
    ],
```

Declaration:
```
{
    "class": "ADC",
    "schemaVersion": "3.7.0",
    "controls": {
        "traceResponse": true
    },
    "tenant": {
        "class": "Tenant",
        "application": {
            "class": "Application",
            "vip": {
                "class": "Service_HTTP",
                "virtualAddresses": [
                    "192.0.2.3"
                ],
                "virtualPort": 443,
                "pool": "pool"
            },
            "pool": {
                "class": "Pool",
                "members": [ ]
            }
        }
    }
}
```

**Comment by dstokesf5** [comment]
Added to our internal product backlog as AUTOTOOL-2522.

**Comment by dstokesf5** [comment]
@0xHiteshPatel Sorry, that follow up issue got lost. I created #465 so we can more reliably track it.

**Comment by dstokesf5** [comment]
This issue has been resolved by the 3.29.0 release.
